pkg/controller/config-controller.go

```diff
@@ -375,54 +376,52 @@ func (r *CDIConfigReconciler) reconcileImportProxy(config *cdiv1.CDIConfig) erro
 		config.Status.ImportProxy.HTTPProxy = &clusterWideProxy.Status.HTTPProxy
 		config.Status.ImportProxy.HTTPSProxy = &clusterWideProxy.Status.HTTPSProxy
 		config.Status.ImportProxy.NoProxy = &clusterWideProxy.Status.NoProxy
-		if clusterWideProxy.Spec.TrustedCA.Name != "" {
-			config.Status.ImportProxy.TrustedCAProxy = &clusterWideProxy.Spec.TrustedCA.Name
-			err = r.reconcileImportProxyCAConfigMap(config, clusterWideProxy)
-			if err != nil {
+		if cmName := clusterWideProxy.Spec.TrustedCA.Name; cmName != "" {
+			if err := r.reconcileImportProxyCAConfigMap(cmName); err != nil {
 				return err
 			}
+			config.Status.ImportProxy.TrustedCAProxy = &cmName
 		}
 	}
 	return nil
 }
 
 // Create/Update a configmap with the CA certificates in the controllor context with the cluster-wide proxy CA certificates to be used by the importer pod
-func (r *CDIConfigReconciler) reconcileImportProxyCAConfigMap(config *cdiv1.CDIConfig, proxy *ocpconfigv1.Proxy) error {
+func (r *CDIConfigReconciler) reconcileImportProxyCAConfigMap(cmName string) error {
 	client := r.uncachedClient
-	cmName := proxy.Spec.TrustedCA.Name
-	if cmName == "" {
-		// Using the default cluster-wide proxy CA certificates configmap name
-		cmName = ClusterWideProxyConfigMapName
-	}
 	clusterWideProxyConfigMap := &v1.ConfigMap{}
-	if err := client.Get(context.TODO(), types.NamespacedName{Name: cmName, Namespace: ClusterWideProxyConfigMapNameSpace}, clusterWideProxyConfigMap); err == nil {
-		// Copy the cluster-wide proxy CA certificates to the importer pod proxy CA certificates configMap
-		if certBytes, ok := clusterWideProxyConfigMap.Data[ClusterWideProxyConfigMapKey]; ok {
-			configMap := &v1.ConfigMap{}
-			if err := client.Get(context.TODO(), types.NamespacedName{Name: common.ImportProxyConfigMapName, Namespace: r.cdiNamespace}, configMap); errors.IsNotFound(err) {
-				proxyConfigMap := r.createProxyConfigMap(certBytes)
-				util.SetRecommendedLabels(proxyConfigMap, r.installerLabels, "cdi-controller")
-				if err := client.Create(context.TODO(), proxyConfigMap); err != nil {
-					return err
-				}
-				return nil
-			}
-			if configMap != nil {
-				configMap.Data[common.ImportProxyConfigMapKey] = certBytes
-				util.SetRecommendedLabels(configMap, r.installerLabels, "cdi-controller")
-				if err := client.Update(context.TODO(), configMap); err != nil {
-					return err
-				}
-			}
+	if err := client.Get(context.TODO(), types.NamespacedName{Name: cmName, Namespace: ClusterWideProxyConfigMapNameSpace}, clusterWideProxyConfigMap); err != nil {
+		return err
```

Comment:
Can we put an event on the cdi cr indicating that the cluster wide proxy doesn't exist in case this error is a NotFound error?

Reply:
Sure, will add.